### ISSUE TITLE
October 2023 Release Status per Platform, Version & Binary Type

### ISSUE THREAD

**Issue by sophia-guo** [adoptium/temurin#6]
Sharing information in this issue since the TCK work is being tracked in temurin-compliance private repo not visible to the community (as per the OCTLA).  Risks and expectations for timing on the release are listed in this [issue comment](https://github.com/adoptium/adoptium/issues/3#issuecomment-866903922).  Primary platforms (x64 Linux/Windows/OSX and aarch64 Linux/OSX) in **bold** are prioritized, secondary platforms not in bold follow in no particular order (as machine resources are available).  We retrospectively measure and track how well we do against these targets in these [Adoptium Release Scorecards](https://github.com/adoptium/adoptium/wiki/Adoptium-Release-Scorecards) in order to continuously assess and improve.

✔️ results in these Tables means the activity has successfully completed.

⏳ results means that we are actively working on closing off the runs needed for this version, platform, binaryType.

⛔ means there is no build planned for that version/platform combination.

⏸️ means activity not yet started.

### JDK8u392-b08
| Platform  |[ jdk8 AQA](https://github.com/adoptium/aqa-tests/issues/4821)  | jdk8 TCK  | jdk8 published | jdk8 installers  | jdk8 images   | Notes |
| -----     | -----     | -----     | -----          | -----            | -----         | ----- |
| **x64 Linux** | ✔️     | ✔️         | ✔️              | ✔️                | ✔️            |       |
| **x64 Windows** | ✔️   | ✔️         | ✔️           | ✔️               | ✔️             |       |
| **x64 Mac** | ✔️      | ✔️        | ✔️             | ✔️               | ⛔            |       |
| **aarch64 Linux** | ✔️     | ✔️ | ✔️            | ✔️                |✔️             |       |
| ppcle64 Linux | ✔️      | ✔️        | ✔️             |✔️               | ✔️             |       |
| ppc64 AIX | ✔️         |  ✔️        | ✔️              | ⛔               | ⛔            |       |
| x32 Windows | ✔️      | ✔️        | ✔️              | ⛔               | ✔️             |       |
| arm32 Linux | ✔️     | ✔️         | ✔️              | ✔️                | ✔️             |       |
| x64 alpine-Linux | ✔️   | ✔️         |  ✔️              | ✔️               | ✔️            | This will be a headless build |
| sparcv9 solaris | ✔️   | ✔️         | ✔️              | ⛔               | ⛔            |       |
| x86 solaris | ✔️     | ✔️         | ✔️            | ⛔               | ⛔            |       |

### jdk-11.0.21+9
| Platform | [jdk11 AQA ](https://github.com/adoptium/aqa-tests/issues/4823)| jdk11 TCK  | jdk11 published| jdk11 installers | jdk11 images  | Notes |
| -----    | -----     | -----      | -----          | -----            | -----         | ----- |
| **x64 Linux** | ✔️    | ✔️          | ✔️             |  ✔️                | ✔️             |       |
| **x64 Windows** | ✔️   | ✔️         | ✔️              | ✔️              | ✔️             |       |
| **x64 Mac** |  ✔️      | ✔️         | ✔️             | ✔️              | ⛔            |       |
| **aarch64 Linux** | ✔️   | ✔️         | ✔️           |  ✔️                | ✔️             |       |
| **aarch64 Mac** | ✔️       | ✔️         | ✔️              | ✔️              | ⛔            |       |
| ppcle64 Linux | ✔️     | ✔️      | ✔️             | ✔️                | ✔️            |       |
| s390x Linux   | ✔️    | ✔️         | ✔️              | ✔️                | ✔️             |       |
| ppc64 AIX | ⏸️         | ⏸️         | ⏸️              | ⛔               | ⛔            |  [Blocked](https://github.com/adoptium/aqa-tests/issues/4683) |
| x32 Windows | ✔️       | ✔️         | ✔️              | ✔️                | ⛔            |       |
| arm32 Linux | ✔️      | ✔️         | ✔️              |  ✔️                | ✔️            |       |
| x64 alpine-Linux | ✔️   | ✔️         | ✔️           | ✔️               | ✔️            | This will be a headless build |

### jdk-17.0.9+9
| Platforms | [jdk17 AQA](https://github.com/adoptium/aqa-tests/issues/4826) | jdk17 TCK | jdk17 published| jdk17 installers | jdk17 images | Notes |
| -----     | -----     | -----     | -----          | -----            | -----        | ----- |
| **x64 Linux** | ✔️    |✔️        | ✔️           |  ✔️                | ✔️            |       |
| **x64 Windows** | ✔️    | ✔️         | ✔️              | ✔️               | ✔️            |  [aqa-tests#4831](https://github.com/adoptium/aqa-tests/issues/4831) |
| **x64 Mac** | ✔️      | ✔️         | ✔️              | ✔️                | ⛔           |       |
| **aarch64 Linux** | ✔️     | ✔️         | ✔️             |  ✔️               | ✔️           |       |
| **aarch64 Mac** | ✔️       | ✔️         | ✔️             | ✔️               | ⛔           |       |
| ppcle64 Linux | ✔️     | ✔️        |✔️             | ✔️                | ✔️            |       |
| s390x Linux   | ✔️     | ✔️         | ✔️               | ✔️                | ✔️            |       |
| ppc64 AIX | ⏸️         | ⏸️         | ⏸️              | ⛔               | ⛔           | [Blocked](https://github.com/adoptium/aqa-tests/issues/4683)      |
| x32 Windows | ✔️        |✔️        | ✔️                 | ✔️                | ⛔           |       |
| arm32 Linux | ✔️       | ✔️         | ✔️             |  ✔️                | ✔️           |       |
| x64 alpine-Linux |✔️  | ✔️         |✔️              | ✔️              | ✔️            | This will be a headless build |

### JDK21.0.1+12
| Platform  | [jdk21 AQA](https://github.com/adoptium/aqa-tests/issues/4827) | jdk21 TCK | jdk21 published| jdk21 installers | jdk21 images  | Notes |
| -----     | -----     | -----     | -----          | -----            | -----         | ----- |
| **x64 Linux** |✔️    | ✔️         | ✔️              | ✔️                | ✔️            |       |
| **x64 Windows** | ✔️   |✔️         | ✔️              | ✔️               | ✔️             | [aqa-tests#4831](https://github.com/adoptium/aqa-tests/issues/4831) |
| **x64 Mac** | ✔️       |✔️         | ✔️              | ✔️                | ⛔            |       |
| **aarch64 Linux** | ✔️    |✔️         | ✔️              |  ✔️                | ✔️             |       |
| **aarch64 Mac** | ✔️      | ✔️         | ✔️              | ✔️             | ⛔           |       |
| ppcle64 Linux | ✔️     |  ✔️         |✔️              | ✔️                | ✔️            |       |
| s390x Linux   | ⛔   | ⛔       | ⛔           | ⛔              | ⛔            |   Not in plan    |
| ppc64 AIX |⛔         | ⛔        | ⛔            | ⛔               | ⛔            |   Not in plan    |
| arm32 Linux | ⛔      | ⛔         | ⛔             | ⛔                | ⛔            |   Not in plan    |
| x64 alpine-Linux | ✔️  | ✔️         | ✔️              |  ✔️               | ✔️             | This will be a headless build |
| aarch64 alpine-Linux | ✔️  | ✔️         | ✔️              | ✔️               | ✔️             | This will be a headless build |

**Comment by Mooninaut** [comment]
The column headers under "JDK21.0.XX+Y ????" say "jdk20" instead of "jdk21".

**Comment by jerboaa** [comment]
> The column headers under "JDK21.0.XX+Y ????" say "jdk20" instead of "jdk21".

Thanks, fixed.

**Comment by Siedlerchr** [comment]
JDK 21.0.1 is now also officially released https://www.oracle.com/java/technologies/javase/21-0-1-relnotes.html

**Comment by rec-bln** [comment]
When can we expect JDK 17.0.9 for windows x64?

**Comment by sophia-guo** [comment]
@rec-bln it's blocked by https://github.com/adoptium/aqa-tests/issues/4831, which we are working on right now.

**Comment by wendigo** [comment]
Is there a plan to release JDK 21 ppc64 build?

**Comment by h-vetinari** [comment]
> Is there a plan to release JDK 21 ppc64 build?

See https://github.com/adoptium/adoptium-support/issues/917 (TL;DR: maybe)

**Comment by rec-bln** [comment]
> @rec-bln it's blocked by [adoptium/aqa-tests#4831](https://github.com/adoptium/aqa-tests/issues/4831), which we are working on right now.

thanks! Looking forward to get the release :) thanks for your work on this!

**Comment by sxa** [comment]
> Is there a plan to release JDK 21 ppc64 build?

Do you mean aix/ppc64 or Linux/ppc64le? The former is a definite no for now as some of the new features are not yet implemented on AIX. Linux/ppc64le has been shipped now.

**Comment by wendigo** [comment]
@sxa `Linux/ppc64le` :)

**Comment by chadlwilson** [comment]
Hi folks - The Windows x64 builds seem to be separated into a separate release at https://github.com/adoptium/temurin17-binaries/releases/tag/jdk-17.0.9%2B9.1 , I guess due to https://github.com/adoptium/aqa-tests/issues/4831 - are these going to be merged into the same GitHub release as everything else or is this the permanent situation for this release?

**Comment by sxa** [comment]
It will be permanent to avoid any confusion for people who picked up the old one and may have a record of the SHAs etc. If you pull from the API you should receive the latest version that is available.

**Comment by chadlwilson** [comment]
@sxa the x64 binaries were never published to the old release though.

This is a bit unfortunate. We didn't use the APIs - IIRC due to rate limiting (not sure if relevant now). and 1 less dependency on external infrs. Guess will need to migrate.

**Comment by huczas** [comment]
During update, there is a problem with package version under ubuntu 2004, 
```
curl https://packages.adoptium.net/artifactory/deb/pool/main/t/temurin-11/temurin-11-jdk_11.0.21.0.0+9_amd64.deb                                             
{
  "errors" : [ {
    "status" : 404,
    "message" : "File not found."
  } ]
}
```
Also output from apt-get during packer installation:
```
 azure-arm: Err:1 https://packages.adoptium.net/artifactory/deb focal/main amd64 temurin-11-jdk amd64 11.0.21.0.0+9
    azure-arm:   400  Bad Request [IP: 146.75.123.42 443]
azure-arm: E: Failed to fetch [https://packages.adoptium.net/artifactory/deb/pool/main/t/temurin-11/temurin-11-jdk_11.0.21.0.0+9_amd64.deb](https://packages.adoptium.net/artifactory/deb/pool/main/t/temurin-11/temurin-11-jdk_11.0.21.0.0+9_amd64.deb)  400  Bad Request [IP: 146.75.123.42 443]
```

**Comment by chadlwilson** [comment]
@sxa Unfortunately the API doesn't really work cleanly either, as to get a deterministic version (compulsory in order to do reproducible builds when repackaging Temurin), it still requires the caller to be aware of the `.1` special case for Windows, since the relevant parameter is the adoptium release name, not the canonical version:

**404s**: https://api.adoptium.net/v3/binary/version/jdk-17.0.9%2B9/windows/x64/jre/hotspot/normal/eclipse?project=jdk
**works:** https://api.adoptium.net/v3/binary/version/jdk-17.0.9%2B9.1/windows/x64/jre/hotspot/normal/eclipse?project=jdk

**Comment by andrew-m-leonard** [comment]
> @sxa Unfortunately the API doesn't really work cleanly either, as to get a deterministic version (compulsory in order to do reproducible builds when repackaging Temurin), it still requires the caller to be aware of the `.1` special case for Windows, since the relevant parameter is the adoptium release name, not the canonical version:
> 
> **404s**: https://api.adoptium.net/v3/binary/version/jdk-17.0.9%2B9/windows/x64/jre/hotspot/normal/eclipse?project=jdk **works:** https://api.adoptium.net/v3/binary/version/jdk-17.0.9%2B9.1/windows/x64/jre/hotspot/normal/eclipse?project=jdk

@chadlwilson You can use "latest", try:
https://api.adoptium.net/v3/binary/latest/17/ga/windows/x64/jre/hotspot/normal/eclipse?project=jdk

**Comment by chadlwilson** [comment]
@andrew-m-leonard Thanks, but `latest` is not deterministic and doesn't allow for reproducible builds. [GoCD](https://github.com/gocd/gocd) repackages and redistributes Temurin JREs as part of a bigger package. Aside from that, using the GitHub URLs directly allows pulling the sha256sums to validate the integrity without intercepting the `30x` from the API call, so there is an additional advantage on top of using the API. Will just hack for this release.

**Comment by sophia-guo** [comment]
@huczas updated. temurin-11-jdk_11.0.21.0.0+9_amd64.deb should be there now. Could you try it again? Thanks!

**Comment by andrew-m-leonard** [comment]
> @andrew-m-leonard Thanks, but `latest` is not deterministic and doesn't allow for reproducible builds. [GoCD](https://github.com/gocd/gocd) repackages and redistributes Temurin JREs as part of a bigger package. Aside from that, using the GitHub URLs directly allows pulling the sha256sums to validate the integrity without intercepting the `30x` from the API call, so there is an additional advantage on top of using the API. Will just hack for this release.

Thanks for the feedback @chadlwilson we will bear that scenario in mind for next time.

**Comment by sxa** [comment]
> @sxa Unfortunately the API doesn't really work cleanly either, as to get a deterministic version (compulsory in order to do reproducible builds when repackaging Temurin), 

Yep I do understand the concern and it has to be balanced with the ability for us to respin with platform-specific fixes without changing the artefacts which have already been uploaded which has caused us some problems in the past. The way we've implemented the rebuilds by giving it a new identifier is generally the safest option.

> it still requires the caller to be aware of the .1 special case for Windows, since the relevant parameter is the adoptium release name, not the canonical version:

Unfortunately yes that is correct if you require full reproducibility of a specific build.

**Comment by huczas** [comment]
> @huczas updated. temurin-11-jdk_11.0.21.0.0+9_amd64.deb should be there now. Could you try it again? Thanks!

Thank you for fast fix, it't working now :-)

**Comment by sophia-guo** [comment]
All done, close this one.
